### PR TITLE
Create & update PolicyReport resources from Insights reports

### PR DESCRIPTION
**Pull request by zlayne**
**Related Issue:**  open-cluster-management/backlog#8621

### Description of changes
- Loop through the Insights response to create PolicyReport resources
- If PolicyReports are resolved update the resource to reflect the resolution

### CODE REVIEW

File: pkg/retriever/reportretriever.go [hunk 1]
```diff
@@ -205,145 +201,33 @@ func (r *Retriever) CallInsights(req *http.Request, cluster types.ManagedCluster
...
-					errorMap[key] = val
-				}
-			}
-			lock.Lock()
-			contentsMap[errorName] = errorMap
-			lock.Unlock()
+			return types.ProcessorData{
+				ClusterInfo: cluster,
+				Reports: reports,
+				Contents: contentsMap,
```
Comment (zlayne) [inline pkg/retriever/reportretriever.go:228]: @xrajesh Im not sure how else to get the contents to the processor, but I dont really like the idea of passing the entire contents object every time. Suggestions?

Comment (xrajesh) [reply pkg/retriever/reportretriever.go:228]: @zlayne the contents Map is going to be static , and periodically refreshed . We dont need to pass every time .  We can export that map, and we could access in reportprocessor.go

File: pkg/processor/reportprocessor.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+// CreateUpdatePolicyReports ...
+func (p *Processor) CreateUpdatePolicyReports(input chan types.ProcessorData) {
+	data := <-input
+	// Loop through Report array and return a PolicyReport for each violation
+	for _, report := range data.Reports.Reports {
+		// Find the correct Insight content data from cache
+		reportData := retriever.ContentsMap[report.Key]
+		if reportData != nil {
```
Comment (xrajesh) [inline pkg/processor/reportprocessor.go:45]: Lets log a message and may be trigger the GET contents to refresh the contentsMap

File: pkg/processor/reportprocessor.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+			reportDataBytes, _ := json.Marshal(reportData)
+			unmarshalError := json.Unmarshal(reportDataBytes, &contentData)
+			if unmarshalError != nil {
+				glog.Infof(
+					"Error unmarshalling Report %v for cluster %s (%s)",
+					unmarshalError,
+					data.ClusterInfo.Namespace,
+					data.ClusterInfo.ClusterID,
+				)
+			}
```
Comment (xrajesh) [inline pkg/processor/reportprocessor.go:56]: If there is an error we have to skip.

File: pkg/processor/reportprocessor.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+				resp.Error(),
+			)
+		} else {
+			glog.Infof(
+				"Successfully updated PolicyReport %s status to error for cluster %s",
+				report.Component,
+				cluster.Namespace,
+			)
+		}
+	}
```
Comment (xrajesh) [inline pkg/processor/reportprocessor.go:173]: Need to handle the case where the status is error and we needed to update other field.

File: pkg/processor/reportprocessor.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+			Do(context.TODO())
+
+		respBytes, _ := getResp.Raw()
+		var prResponse types.PolicyReportGetResponse
+		unmarshalError := json.Unmarshal(respBytes, &prResponse)
+		if unmarshalError != nil {
+			glog.Infof(
+				"Error unmarshalling PolicyReport: %v",
+				unmarshalError,
+			)
```
Comment (xrajesh) [inline pkg/processor/reportprocessor.go:196]: We may want to skip if there is an errror

**Pull request merged**